pkg/eval/builtin_fn_flow.go

```diff
@@ -50,7 +50,73 @@ import (
 
 // TODO(xiaq): Document "multi-error".
 
-// TODO(xiaq): Document "return", "break" and "continue".
+//elvdoc:fn break
+//
+// Raise the special "break" exception. Loops capture this exception and cause
+// the loop to terminate. Because it is an exception it can be caught by a
+// [`try`](language.html#exception-control-try) block. If not caught, either
+// implicitly by a loop or explicitly, it causes a failure like any other
+// uncaught exception.
+//
+// See the discussion about [flow commands and exceptions](language.html#exception-and-flow-commands)
+//
+// **Note**: You can create a `break` function and it will shadow the builtin
+// command. If you do so you should explicitly invoke the builtin. For example:
+//
+// ```elvish-transcript
+// > fn break []{ put break; builtin:break; put 'should not appear' }
+// > for x [a b c] { put $x; break }
+// ▶ a
+// ▶ break
+// ```
+
+//elvdoc:fn continue
+//
+// Raise the special "continue" exception. Loops capture this exception and
+// cause the loop to begin its next iteration. Because it is an exception it
+// can be caught by a [`try`](language.html#exception-control-try) block. If
+// not caught, either implicitly by a loop or explicitly, it causes a failure
+// like any other uncaught exception.
+//
+// See the discussion about [flow commands and exceptions](language.html#exception-and-flow-commands)
+//
+// **Note**: You can create a `continue` function and it will shadow the builtin
+// command. If you do so you should explicitly invoke the builtin. For example:
+//
+// ```elvish-transcript
+// > fn break []{ put continue; builtin:continue; put 'should not appear' }
+// > for x [a b c] { put $x; break }
+// ▶ a
+// ▶ continue
+// ▶ b
+// ▶ continue
+// ▶ c
+// ▶ continue
+// ```
+
+//elvdoc:fn return
+//
+// Raise the special "return" exception. Functions capture this exception and
+// cause the function to terminate. Because it is an exception it can be
+// caught by a [`try`](language.html#exception-control-try) block. If not
+// caught, either implicitly by a function or explicitly, it causes a failure
+// like any other uncaught exception.
+//
+// See the discussion about [flow commands and exceptions](language.html#exception-and-flow-commands)
+//
+// **Note**: You can create a `return` function and it will shadow the builtin
+// command. **Do not do this!**. You cannot propagate the exception to the
+// calling function. In this example you might, incorrectly, expect "no" to
+// not appear in the output:
+//
+// ```elvish-transcript
+// > fn return []{ put return; builtin:return; put 'should not appear' }
+// > fn test-return []{ put yes; return; put no }
+// > test-return
+// ▶ yes
+// ▶ return
+// ▶ no
+// ```
```

Comment:
This example really shows two things: (1) the fact that you can shadow the builtin `return`, and that (2) you can't create a wrapper for the `return` command using a `fn`-defined function, since the inner function itself will capture the exception. It's better to break this into two examples.

Reply:
That is true. Although probably not in the fashion you expect. Using a lambda results in an infinite loop:

```
> return~ = []{ put return; builtin:return; put 'should not appear' }
> fn test-return []{ put yes; return; put no }
> test-return
▶ yes
▶ return
▶ return
▶ return
[ad infinitum]
```

That seems like a bug. Nonetheless, I don't see the point in creating two examples since this is a "**do not do this**" scenario as explained by the preceding paragraph.